src/components/LoginButton.jsx

```diff
@@ -1,37 +1,59 @@
 import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
 import firebase from 'firebase';
+import { CircularProgress, makeStyles } from '@material-ui/core';
 import Button from '@material-ui/core/Button';
 import ProfileMenu from './ProfileDropdown';
 
+const useStyles = makeStyles(() => ({
+  progress: {
+    display: 'block',
+    margin: 'auto',
+  },
+}));
+
 export default function LoginButton() {
   const [isSignedIn, setIsSignedIn] = useState(!!firebase.auth().currentUser);
+  const [loading, setLoading] = useState(true);
+  const [error] = useState();
+  const classes = useStyles();
+
+  function updateUserStatus(user) {
+    setLoading(false);
+    setIsSignedIn(!!user);
+  }
 
   useEffect(
-    () => firebase.auth().onAuthStateChanged((user) => setIsSignedIn(!!user)),
+    () => firebase.auth().onAuthStateChanged((user) => updateUserStatus(user)),
     []
   );
 
   const signUserOut = () => firebase.auth().signOut();
   const user = firebase.auth().currentUser;
 
-  return isSignedIn ? (
-    <ProfileMenu
-      signOut={signUserOut}
-      name={user.displayName}
-      email={user.email}
-      photo={user.photoURL}
-    />
-  ) : (
-    <Button
-      variant="contained"
-      color="primary"
-      component={Link}
-      to={{
-        state: { showLoginDialog: true },
-      }}
-      replace>
-      Login
-    </Button>
+  return (
+    <>
+      {error?.toString() ||
+        (loading && <CircularProgress className={classes.progress} />) ||
```

Comment:
This isn't a very smooth transition. I think rather than this we can show nothing when the auth state is unknown and then use a transition like [Zoom](https://material-ui.com/components/transitions/#zoom) to display whatever element is appropriate

Reply:
Yeah that does make a lot of sense. I just changed the CircularProgress component with a null component (showing nothing) now I will work on getting a Zoom display for the transition of the avatar